### ISSUE TITLE
Replacing long text with composable input has a visible delay

### ISSUE THREAD

**Issue by f1ames** [ckeditor/ckeditor5-typing#145]
Found while working on https://github.com/ckeditor/ckeditor5-engine/issues/403.

While `diff` in mutations is used only for simple mutations (inserting single character on collapsed or non-collapsed selection) it works fine for regular typing, but when character is composable one (e.g. Spanish accent or Hiragana typing) the [`diff` function used in mutations](https://github.com/ckeditor/ckeditor5-typing/blob/5a1ff55af4c053c6020addf6cb1a0c55cee3ece3/src/input.js#L276) slows down significantly:

![diff3](https://user-images.githubusercontent.com/1061942/38622261-10bc7c1e-3da3-11e8-983f-4fef3b327c33.gif)

**Comment by Reinmar** [comment]
Cool. Good to know that we can improve this if we'll introduce a fast diff.

**Comment by f1ames** [comment]
Yes, we should check if [`fastDiff`](https://github.com/ckeditor/ckeditor5-utils/issues/235) (when merged to `master`) can be used here (in mutations handler) and if it can improve this situation.

**Comment by Reinmar** [comment]
Still todo?

**Comment by f1ames** [comment]
Yes, since `fastDiff` was already merged we may start working on this one. It will require switching from `diff` to `fastDiff` in mutations handler.

The only difference is that `diff` is more accurate. It finds (well, in theory at least) minimal number of changes, while `fastDiff` is basically a partial diff so number of _insertions_ and _deletions_ will usually be higher. Still it shouldn't be a problem here (we should check if it will has some effect on undo steps).
